Src/FluentAssertions/Json/JsonElementAssertions.cs

```diff
@@ -0,0 +1,152 @@
+    /// </summary>
+    /// <param name="subject">The subject.</param>
+    public JsonElementAssertions(JsonElement subject)
+    {
+        Subject = subject;
+    }
+
+    /// <summary>
+    ///     Asserts that the current <see cref="JsonElement"/> has the specified <see cref="JsonValueKind"/>.
+    /// </summary>
+    /// <param name="valueKind">The JSON string.</param>
+    /// <param name="because">
+    ///     A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    ///     is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    ///     Zero or more objects to format using the placeholders in <see paramref="because" />.
+    /// </param>
+    public AndConstraint<JsonElementAssertions> HaveValueKind(JsonValueKind valueKind, string because = "", params object[] becauseArgs)
+    {
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(Subject.ValueKind == valueKind)
+            .FailWith("Expected {context:JSON} to have value kind {0}{reason}, but found {1}.", valueKind, Subject.ValueKind);
+
+        return new(this);
+    }
+
+    /// <summary>
+    ///     Asserts that the current <see cref="JsonElement"/> is the JSON null node.
+    /// </summary>
+    /// <param name="because">
+    ///     A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    ///     is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    ///     Zero or more objects to format using the placeholders in <see paramref="because" />.
+    /// </param>
+    public AndConstraint<JsonElementAssertions> BeNull(string because = "", params object[] becauseArgs)
+    {
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(Subject.ValueKind == JsonValueKind.Null)
+            .FailWith("Expected {context:JSON} to be a JSON null{reason}, but found {0}.", Subject);
+
+        return new(this);
+    }
+
+    /// <summary>
+    ///     Asserts that the current <see cref="JsonElement"/> is the JSON string node.
+    /// </summary>
+    /// <param name="value">The value of the JSON string node.</param>
+    /// <param name="because">
+    ///     A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    ///     is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    ///     Zero or more objects to format using the placeholders in <see paramref="because" />.
+    /// </param>
+    public AndConstraint<JsonElementAssertions> BeString(string value, string because = "", params object[] becauseArgs)
```

Comment:
Why not use `Be()` in every case? The type info on the method name is redundant.

Reply:
I considered that, and if we would only assert on simple nodes, that might would be sufficient.

However, if the JSON is a complex structure, you might want to assert against the JSON string it results in, what is different then the a JSON string node/element. Besides that, it also makes implicit that both the value and the element type are asserted.

In general, what the API should look like for JSON array, and JSON object results, is something I'm still thinking about. I think I personally would use this to assert on the full JSON result as a string:

``` C#
options.Should().Serialize(myComplexModel)
    And.Shoud().Be("{ 'prop': 'Value', 'other': 12, children: [ ]}");
```

but others might opt for:


``` C#
options.Should().Serialize(myComplexModel)
    And.Shoud().Be(new()
    {
        prop = "Value" ,
        other = 12,
        children = Array.Empty<object>()
     });
```

Where the anonymous object is used to represent the `JsonElement`. The latter is normally used for `BeEquivalent()`, but I'm wondering if that has meaning for this kind of assertions.

I'm also wondering if we want the opposite of `BeString()`, etc. Because, what does that say? That I except something not to be a string? Or a string, but with a different value?!